### PR TITLE
Update PowerShell to build with .NET Core 2.0.4

### LINKED ISSUE #5428: Use dotnetcore 2.0.4 final for GA
Currently we're using a preview release of 2.0.4 servicing.  We need to pick up their final release for GA.

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Close #5428
Update PowerShell to build with .NET Core 2.0.4. (turn out to be a simple change 😄)

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [na] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [na] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [na] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [x] [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Review by anmenaga — APPROVED** [review]
LGTM

File: PowerShell.Common.props [hunk 1]
```diff
@@ -94,7 +94,7 @@
     <Copyright>(c) Microsoft Corporation. All rights reserved.</Copyright>
 
     <TargetFramework>netcoreapp2.0</TargetFramework>
-    <RuntimeFrameworkVersion>2.0.4-servicing-25831-01</RuntimeFrameworkVersion>
```
Comment (TravisEz13) [inline PowerShell.Common.props]: should we update the appveyor yml to flush the cache?

**Review by TravisEz13 — APPROVED** [review]

File: PowerShell.Common.props (hunk 1 above)
Comment (daxian-dbw) [reply PowerShell.Common.props]: Good point! Fixed.

**Comment by daxian-dbw** [comment]
Investigating the xUnit test failures.

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by rkeithhill** [comment]
Don't forget to update - https://github.com/PowerShell/PowerShell/blob/master/global.json

**Comment by daxian-dbw** [comment]
@rkeithhill That is the SDK version. SDK version is not changed in this PR and we are still using 2.0.2. This PR only changes the version of .NET Core Runtime Framework that powershell is built with.

**Comment by rkeithhill** [comment]
I'm beginning to understand @vors rant on Twitter about .NET Core versioning.  :-)

**Pull request merged**